### ISSUE TITLE
one-to-one impossible to get object

### ISSUE THREAD

**Issue by italien** [satyan/sugar#382]
I use sugarOrm and I want to save a place with an address object, when I save everything is fine but when I try to get address it's always null, the others data from the place are saved and I can get them but I don't know why I can't get my address object and its informations.

My code:

    public class PlaceModel extends SugarRecord<PlaceModel> implements Parcelable {
    @Ignore
    @Expose
    private ArrayList<CategoryModel> categories = new ArrayList<CategoryModel>();
    @Ignore
    @Expose
    private ArrayList<MediaContentModel> medias = new ArrayList<MediaContentModel>();
    @Ignore
    @Expose
    private ArrayList<ReviewContentModel> reviews = new      A ArrayList<ReviewContentModel>();
    @Expose
    private String name;
    @Expose
    @Ignore
    private List<Double> location = new ArrayList<Double>();
    @Expose
    private String website;
    @Expose
    private AddressModel address;
    @SerializedName("global_score")
    @Expose
    private Integer globalScore;
    @SerializedName("popularity_score")
    @Expose
    private Integer popularityScore;
    @SerializedName("quality_score")
    @Expose
    private Integer qualityScore;
    @Expose
    @Ignore
    private Object prices;
    @Expose
    @Ignore
    private Object hours;
    @Expose
    @Ignore
    private Object phone;
    @Expose
    private String createdAt;
    @Expose
    private String updatedAt;
    @SerializedName("idObject")
    @Expose
    private String idPlace;

    private String listCategory;

    private String listMedias;

    private String listReviews;

    private String listLocation;

    public PlaceModel() {
    }
    // Setters, getters and stuff for parcelable
}
Address Model :

    public class AddressModel extends SugarRecord<AddressModel> implements Parcelable {

    @SerializedName("street_address")
    @Expose
    private String streetAddress;
    @SerializedName("city")
    @Expose
    private String city;
    @SerializedName("state")
    @Expose
    @Ignore
    private Object state;
    @SerializedName("postal_code")
    private String postalCode;
    @SerializedName("country")
    @Expose
    private String country;

    public AddressModel(){
    }
    // Setters, getters and stuff for parcelable
}
When I save :

    public void getPlaceDetails(String id)
    {
        RestClient client = RestClient.getInstance();
        propertiesReader = new PropertiesReader(mContext);
        Properties apiProperties = propertiesReader.loadPropertiesFile("api.properties");

        client.createRestClient(apiProperties.getProperty("APIUrl"), mContext);
        ApiManager apiManager = client.getApiManager();
        PlaceModel place = apiManager.getPlaceDetails(id);
        if (place != null) {
            Gson gson = new GsonBuilder().create();
            place.setListCategory(gson.toJsonTree(place.getCategories()).getAsJsonArray().toString());
            place.setListMedias(gson.toJsonTree(place.getMedias()).getAsJsonArray().toString());
            place.setListReviews(gson.toJsonTree(place.getReviews()).getAsJsonArray().toString());
            place.setListLocation(gson.toJsonTree(place.getLocation()).getAsJsonArray().toString());
            places.add(place);
            place.save();
        }
    }
If I try to get address I get a NullPointerException. I followed the doc of sugarOrm, Do I'm missing something ?

**Comment by Andycharalambous** [comment]
I have the same problem - I can save 2 objects and then set one as child of the other using the sugar orm support for one-to-one relationships, but then when I retrieve the parent object later, the child relationship is always null.

E.g.

Parent p = new Parent();
p.save(); // id = 1 now.
Child c = new Child();
c.save();
p.setChild(c);
p.save(); 
p = Parent.findById(Parent.class, 1L);
assert p.getChild() != null // boom!

**Comment by umaso** [comment]
@Andycharalambous ,have you figured it out?I'm facing the same problem.with sugar1.3......the chlid relationship is always null.that's awful.

**Comment by thecodingwizard** [comment]
yes. This needs to be fixed. A very temporary and bad solution would be to store the ID of the Child class and then run another query to get it.
